anni-playback/src/utils/blocking_rb.rs

```diff
@@ -102,8 +105,14 @@ impl<T: Copy + Clone + Default + Sync + Send> BlockingRb<T, Producer> {
             // Wait for the event to tell us that there free space
             // available or that the operation should be cancelled.
             let (mutex, cvar) = &*self.producer_events;
-            let mut event = mutex.lock().unwrap();
-            event = cvar.wait(event).unwrap();
+            let event = mutex.lock().unwrap();
+            let (event, timeout) = cvar
+                .wait_timeout(event, Duration::from_millis(500))
```

Comment:
Let's add a TODO here, pointing to this PR?

Reply:
done